switch.go

```diff
@@ -409,6 +413,34 @@ func (sw *Switch) StopPeerForError(peer *Peer, reason interface{}) {
 	sw.removePeerFromReactors(peer, reason)
 }
 
+// Attempt to reconnect to the peer
+func (sw *Switch) ReconnectPeerOnError(peer *Peer, reason interface{}) {
+	addr := peer.Connection().RemoteAddress
+	sw.StopPeerForError(peer, reason)
+
+	// in a go-routine so we don't recur infinitely
```

Comment:
Why would we recur infinitely? if there are max 30 attempts. The other reason could be not to block the switch because other peers might want to connect/reconnect